### ISSUE TITLE
How to set up aria lable for reading out the rows are being filtered by screenreader

### ISSUE THREAD

**Issue by PavanJagini** [OfficeDev/office-ui-fabric-react#9820]
## Actual behavior:
how to setup aria label for making screenreader users to understand the rows are filtered in detailslist control once done.

**Comment by joschect** [comment]
@PavanJagini Thanks for your question. Let me confirm the issue, it sounds like you are expecting the details list to read out when it has been filtered, is that correct? 

Based on what I'm seeing in the details list code it looks like the filterAriaLabel is applied to the column that's filtered, which means the text will get read out when a user focuses that column. Could you check and see if the text is read out when the column is focused?

**Comment by joschect** [comment]
@PavanJagini I believe that the details list is currently working as expected. 

The details list shouldn’t announce when the filter changes unless it’s being focused directly. Since the filter searchbox isn’t owned by the details list it’s up to that component to provide the details necessary for narrator to announce when the filter is applied. However, narrator should announce the current state of the filter if a user focuses the details list because it will know that it’s currently filtered.

**Comment by eveleens** [comment]
This is related to bug 4 in #6646

I have similar accessibility bug on this: 

**Actual:**
Screen reader does not announce the message displayed on the screen when user search for any invalid value. 


**Expected:**
Screen reader should announce the message displayed on the screen when user search for any invalid value i.e. "No matches found".

**Comment by joschect** [comment]
@eveleens I don't believe that this is related. The screen readers announcement for invalid values is not owned by the details list. The fabric details list doesn't have any knowledge of what the filter is doing. The details list is just given the list of items and told to render those things. It's up to the searchbox to announce all of this information. Or another 3rd component that connects the two of them.

**Comment by eveleens** [comment]
Reading again - agree. Disregard my comment.

**Comment by PavanJagini** [comment]
> @PavanJagini I believe that the details list is currently working as expected.
> 
> The details list shouldn’t announce when the filter changes unless it’s being focused directly. Since the filter searchbox isn’t owned by the details list it’s up to that component to provide the details necessary for narrator to announce when the filter is applied. However, narrator should announce the current state of the filter if a user focuses the details list because it will know that it’s currently filtered.



> @PavanJagini Thanks for your question. Let me confirm the issue, it sounds like you are expecting the details list to read out when it has been filtered, is that correct?
> 
> Based on what I'm seeing in the details list code it looks like the filterAriaLabel is applied to the column that's filtered, which means the text will get readout when a user focuses that column. Could you check and see if the text is read out when the column is focused?

@joschect , Sorry for the delay in the reply. If I go with this approach, autofocusing needs to be done on search box key-up even which will often get the focus out of textbox forcefully in the middle of user typing into the textbox.

Any more thoughts to it?

**Comment by joschect** [comment]
@PavanJagini there are lots of ways that you can implement that. You don't need to autofocus the the details list. You could use an aria-live region to announce the filtered results, or to announce that the results have been filtered.
